AprilSylph/XKit-Rewritten
# No Recommended: Fix blog carousels inconsistency with pagination

I do not like this, Sam I Am.

### Description


In order to exclude the blog carousels at the tops of the blog subs and favs/crushes dashboards, which do not contain recommended blogs, from being hidden by No Recommended, we excluded any carousel not preceded by a post. This, however, means that recommended blog carousels that are coincidentally at the top of any given timeline due to pagination will not be hidden.

Detecting pagination details in redpop is extremely hard unless I am missing something, and the elements in question have somewhat varying titles, so the only way I can think to distinguish between them is that the carousels we want to keep are preceded by "announcements" and the ones we want to hide are preceded by "titles."

This only hides carousels preceded by elements using the `titleObject` css, relying on the styling of the blog subs and favs carousels not changing to be consistent with the recommendation carousel.

To not rely on css, but still rely on the title elements being internally different, one can instead query `(await timelineObject(listTimelineObject.previousElementSibling)).objectType` for `'announcement'` or `'title'`.

Resolves #812.

### Testing steps


- Optionally revert the reversion of the test CSS.
- Enable No Recommended and "Hide recommended blogs between posts"
- Check https://www.tumblr.com/dashboard/blog_subs, https://www.tumblr.com/timeline/blog_subscriptions, https://www.tumblr.com/dashboard/crushes, and https://www.tumblr.com/timeline/crushes. Ensure no carousels are hidden
- Check the dashboard and a tagged page with endless scrolling disabled. Ensure carousels are hidden.
